projects/angular-auth-oidc-client/src/lib/validation/jsrsasign-reduced.service.ts

```diff
@@ -1,20 +1,50 @@
 import { Injectable } from '@angular/core';
-import { hextob64u, KJUR } from 'jsrsasign-reduced';
+import { from, Observable } from 'rxjs';
+import { map } from 'rxjs/operators';
 
 @Injectable()
 export class JsrsAsignReducedService {
-  generateCodeChallenge(codeVerifier: any): string {
-    const hash = KJUR.crypto.Util.hashString(codeVerifier, 'sha256');
-    const testData = hextob64u(hash);
+  private crypto: Crypto = window.crypto || (window as any).msCrypto; // for IE11
```

Comment:
This (and the other global `window` calls) break support of Angular Universal (SSR).

The correct way to use window, is by injecting the document `@Inject(DOCUMENT) private readonly doc: Document,` and then using it `this.doc.defaultView`.

Also, since this version is for Angular 13, and the support  of Internet Explorer is deprecated by the Angular team, i don't think it makes sense to consider IE11.

Reply:
That is absolutely correct. We already use the injection of the document in other places. We will fix that in a separate PR. Thanks for the hint!

Reply:
No problem. We collaborate to help each other :) Thanks to you guys!

Reply:
https://github.com/damienbod/angular-auth-oidc-client/issues/1316